run/run.py

```
@@ -0,0 +1,348 @@
+
+1. Copy run/running.example.ini to run/running.ini
+2. Modify the applicable fields of run/running.ini (this may also involve installing browsers)
+3. Make sure you have the correct secret in run/running.ini
+4. Install dependencies with `pip3 install -r requirements.txt`
+5. Make sure you have gsutil installed (see https://cloud.google.com/storage/docs/gsutil)
+
+The script will only accept platform IDs listed in browsers.json.
+
+By default this script will not upload anything! To do a full run for production:
+
+    ./run/run.py firefox-56.0-linux --upload --create-testrun
+
+# Filesystem and network output
+
+- This script will only write files under config['build_path']
+- One run will write approximately 111MB to the filesystem
+- If --upload is specified, it will upload that 111MB of results to Google Storage
+- To upload results, you must be logged in with `gcloud` and authorized
+"""
+
+def main(platform_id, platform, args, config):
+    assert sys.version_info.major == 3, 'This script requires Python 3.'
+
+    print('PLATFORM_ID:', platform_id)
+    print('PLATFORM INFO:', platform)
+
+    if args.path:
+        print('Running tests in path: %s' % args.path)
+    else:
+        print('Running all tests!')
+
+    if args.upload:
+        print('Setting up storage client')
+        from google.cloud import storage
+        storage_client = storage.Client(project='wptdashboard')
+        bucket = storage_client.get_bucket(config['gs_results_bucket'])
+        verify_gsutil_installed(config)
+
+    if args.create_testrun:
+        assert len(config['secret']) == 64, 'Valid secret required to create TestRun'
+
+    if platform['browser_name'] == 'chrome':
+        browser_binary = config['chrome_binary']
+        webdriver_binary = config['chromedriver_binary']
+    elif platform['browser_name'] == 'firefox':
+        browser_binary = config['firefox_binary']
+        webdriver_binary = config['geckodriver_binary']
+    else:
+        raise
```

Might want to raise a more specific exception, or `assert False, "Unsupported browser"`.